src/SerilogTracing/Instrumentation/HttpHandlerDiagnosticObserver.cs

```diff
@@ -0,0 +1,71 @@
+using System.Diagnostics;
+using Serilog.Events;
+using Serilog.Parsing;
+using SerilogTracing.Interop;
+
+namespace SerilogTracing.Instrumentation;
+
+sealed class HttpHandlerDiagnosticObserver : IObserver<KeyValuePair<string,object?>>
+{
+    static readonly Func<object, HttpRequestMessage?> GetRequest = CreateAccessor<HttpRequestMessage>(
+        "System.Net.Http.DiagnosticsHandler+ActivityStartData, System.Net.Http", "Request");
+    
+    static readonly Func<object, HttpResponseMessage?> GetResponse = CreateAccessor<HttpResponseMessage>(
+        "System.Net.Http.DiagnosticsHandler+ActivityStopData, System.Net.Http", "Response");
+
+    static readonly MessageTemplate MessageTemplateOverride =
+        new MessageTemplateParser().Parse("HTTP {RequestMethod} {RequestUri}");
+    
+    public void OnCompleted()
+    {
+    }
+
+    public void OnError(Exception error)
+    {
+    }
+
+    public void OnNext(KeyValuePair<string, object?> value)
+    {
+        if (value.Value == null || Activity.Current == null) return;
+        var activity = Activity.Current;
+        
+        if (value.Key == "System.Net.Http.HttpRequestOut.Start" &&
+            activity.OperationName == "System.Net.Http.HttpRequestOut")
+        {
+            var request = GetRequest(value.Value);
+            if (request == null || request.RequestUri == null) return;
+
+            // The message template and properties will need to be set through a configurable enrichment
+            // mechanism, since the detail/information-leakage trade-off will be different for different
+            // consumers.
+            
+            // For now, stripping any user, query, and fragment should be a reasonable default.
+
+            var uriBuilder = new UriBuilder(request.RequestUri)
```

Comment:
Would it be better to construct a new blank Uri here using just the scheme, host, and path of `request.RequestUri` instead of picking fields to null out?

Reply:
Needs handling of default ports, IIRC.

I went this way because it fits the "redaction" theme, but not really committed one way or the other.